### PR TITLE
hard to say that but it is try to refactor

### PR DESCRIPTION
**Pull request by Nick46368**

### CODE REVIEW

File: Planes/experimentalPlane.js [hunk 1]
```diff
@@ -1,68 +1,35 @@
 const Plane = require('./Plane');
 
-class experimentalPlane   extends Plane
+class ExperimentalPlane extends Plane
 {
-
-    constructor(model, maxSpeed, maxFlightDistance, maxLoadCapacity, type, classificationLevel)  {
+    constructor(model, maxSpeed, maxFlightDistance, maxLoadCapacity, type, classificationLevel) {
```
Comment (DenysLepekha) [inline Planes/experimentalPlane.js:5]: Same here.

File: test/airport.spec.js [hunk 2]
```diff
@@ -1,42 +1,18 @@
...
-        new experimentalPlane("Bell X-14", 277, 482, 500, ExperimentalTypes.HIGH_ALTITUDE, ClassificationLevel.SECRET),
-        new experimentalPlane("Ryan X-13 Vertijet", 560, 307, 500, ExperimentalTypes.VTOL, ClassificationLevel.TOP_SECRET)
-    ];
-    let planeWithMaxPassengerCapacity = new PassengerPlane('Boeing-747', 980, 16100, 70500, 242);
+describe('should test different types of planes', () => {
 
     it('should have military Planes with transport type', () => {
-        let airport = new Airport(planes);
         let transportMilitaryPlanes = airport.getTransportMilitaryPlanes();
         let flag = false;
```
Comment (DenysLepekha) [inline test/airport.spec.js:13]: Use built-in chai assertions e.g. assert.isNotEmpty(transportMilitaryPlanes);

File: test/airport.spec.js [hunk 3]
```diff
@@ -1,42 +1,18 @@
...
-        new MilitaryPlane('B-2 Spirit', 1030, 22000, 70000, MilitaryType.BOMBER),
-        new MilitaryPlane('B-52 Stratofortress', 1000, 20000, 80000, MilitaryType.BOMBER),
-        new MilitaryPlane('F-15', 1500, 12000, 10000, MilitaryType.FIGHTER),
-        new MilitaryPlane('F-22', 1550, 13000, 11000, MilitaryType.FIGHTER),
-        new MilitaryPlane('C-130 Hercules', 650, 5000, 110000, MilitaryType.TRANSPORT),
-        new experimentalPlane("Bell X-14", 277, 482, 500, ExperimentalTypes.HIGH_ALTITUDE, ClassificationLevel.SECRET),
-        new experimentalPlane("Ryan X-13 Vertijet", 560, 307, 500, ExperimentalTypes.VTOL, ClassificationLevel.TOP_SECRET)
-    ];
-    let planeWithMaxPassengerCapacity = new PassengerPlane('Boeing-747', 980, 16100, 70500, 242);
+describe('should test different types of planes', () => {
```
Comment (DenysLepekha) [inline test/airport.spec.js:9]: Message here should be a starting point for the following 'its' similarly as in removed version: describe('My Test', () => {

File: test/airport.spec.js [hunk 4]
```diff
@@ -45,17 +21,14 @@ describe('My Test', () => {
     });
 
     it('should check passenger plane with max capacity', () => {
-        let airport = new Airport(planes);
         let expectedPlaneWithMaxPassengersCapacity = airport.getPassengerPlaneWithMaxPassengersCapacity();
-        assert.isFalse( expectedPlaneWithMaxPassengersCapacity == planeWithMaxPassengerCapacity);
+        assert.isFalse( expectedPlaneWithMaxPassengersCapacity == PassengerPlane);
```
Comment (DenysLepekha) [inline test/airport.spec.js:25]: Passenger plane with max capacity should be checked here.

File: test/airport.spec.js [hunk 5]
```diff
@@ -45,17 +21,14 @@ describe('My Test', () => {
...
     });
 
 
-    it('test 3', () => {
-        console.log("TEST testGetPassengerPlaneWithMaxCapacity started!");
-        let airport = new Airport(planes);
+    it('should sort passenger plane with max load capacity', () => {
```
Comment (DenysLepekha) [inline test/airport.spec.js:29]: Compare instance of PassengerPlane class with expected data and use chai built-in asseretions.

File: test/airport.spec.js [hunk 6]
```diff
@@ -68,9 +41,8 @@ describe('My Test', () => {
         assert.isTrue(nextPlaneMaxLoadCapacityIsHigherThanCurrent);
     })
 
-    it('testHasAtLeastOneBomberInMilitaryPlanes', () => {
-        let airport = new Airport(planes);
-        let bomberMilitaryPlanes  = airport.getBomberMilitaryPlanes ();
+    it('should test that has at least one bomber in military planes', () => {
```
Comment (DenysLepekha) [inline test/airport.spec.js:44]: Use chai built-in assertions. Array isn't empty can be checked here.

File: test/airport.spec.js [hunk 7]
```diff
@@ -80,18 +52,14 @@ describe('My Test', () => {
                 assert.fail("Test failed!");
             }
         }
-
-        // if not failed;
     })
 
-    it('should check that experimentsl planes has classification level higher than unclassified', () => {
-        let airport = new Airport(planes);
+    it('should check that experimental planes has classification level higher than unclassified', () => {
```
Comment (DenysLepekha) [inline test/airport.spec.js:57]: Check and fix this test.

**Review by DenysLepekha — CHANGES_REQUESTED** [review]

File: Airport/Airport.js [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+        let planeWithMaxCapacity = passengerPlanes[0];
+        for (let i = 0; i < passengerPlanes.length; i++) {
+            if (passengerPlanes[i].getPassengersCapacity() > planeWithMaxCapacity.getPassengersCapacity()) {
+                planeWithMaxCapacity = passengerPlanes[i];
+            }
+        }
+        return planeWithMaxCapacity;
+    }
+
+    getTransportMilitaryPlane() {
```
[the remark this answers is not in the document]
Comment (Nick46368) [reply Airport/Airport.js:50]: from 3 to 1

File: test/airport.spec.js (hunk 3 above)
Comment (Nick46368) [reply test/airport.spec.js:9]: return 'My test' as starting point

File: test/airport.spec.js (hunk 4 above)
Comment (Nick46368) [reply test/airport.spec.js:25]: isEqual =

File: test/airport.spec.js (hunk 5 above)
Comment (Nick46368) [reply test/airport.spec.js:29]: take first object of array with max load capacity and compare it

File: test/airport.spec.js [hunk 9]
```diff
@@ -1,101 +1,32 @@
...
-        let airport = new Airport(planes);
-        let expectedPlaneWithMaxPassengersCapacity = airport.getPassengerPlaneWithMaxPassengersCapacity();
-        assert.isFalse( expectedPlaneWithMaxPassengersCapacity == planeWithMaxPassengerCapacity);
+        let actualPlaneWithMaxPassengersCapacity = airport.getPassengerPlaneWithMaxPassengersCapacity();
+        assert.deepEqual(actualPlaneWithMaxPassengersCapacity, planes.filter(plane => plane.model === 'Boeing-747')[0], 'This is not needed plane');
     });
 
+    it('should sort passenger plane with max load capacity', () => {
+        let planesSortedByMaxLoadCapacity = airport.sortPlanesByCharacteristics('maxLoadCapacity');
+        assert.deepEqual(planesSortedByMaxLoadCapacity, planes.sort((a, b)  => (a['maxLoadCapacity'] > b['maxLoadCapacity'] ? 1 : -1)));
```
Comment (DenysLepekha) [inline test/airport.spec.js:20]: Here you compare the logic from the same methods. Take first and last planes from planesSortedByMaxLoadCapacity variable and compare it with objects which contain test data that you expect.

**Review by DenysLepekha — CHANGES_REQUESTED** [review]

**Review by DenysLepekha — APPROVED** [review]

**Pull request merged**